src/storage/src/source/kafka.rs

```diff
@@ -1737,8 +1739,6 @@ fn spawn_metadata_thread<C: ConsumerContext>(
                 if tx.send((probe_ts, update)).is_err() {
                     break;
                 }
-
-                thread::park_timeout(poll_interval);
```

Comment:
Is the change from `park_timeout` to using `sleep` a behavior change? Or can `sleep` also be woken up the same way? Might be worth considering but I haven't looked into the behavior of either function.

Reply:
It is! The difference is that a parked thread can be woken up by another thread, a sleeping thread cannot. I think in this case we don't need the wakeup-by-other-thread behavior and so `sleep` is the right way, but lmk if I missed something!

Reply:
Park/unpark was introduced here: https://github.com/MaterializeInc/materialize/pull/10845, so it was meaningful. Not sure where things are currently, I think the thread now works somewhat differently. But still might be worth it to keep the immediate-cleanup-on-drop behavior.

Reply:
Ah, good to know! Just changing the `sleep` to a `park_timeout` doesn't help much though. The sleep is now in the Ticker which only returns from a tick when the interval is reached, regardless of whether it's woken up before or not. The unpark-on-drop code also doesn't exist anymore.

We could change both of these things, but it doesn't seem worthwhile to pile on complexity just to shut down the thread slightly faster. With RLU the tick interval will be 1s, which is quite different from the 5m interval that existed when the `unpark_timeout` was added.